distro.py

```diff
@@ -654,8 +655,27 @@ def __init__(self,
         self.root_dir = root_dir
         self.etc_dir = os.path.join(root_dir, 'etc') \
             if root_dir else _UNIXCONFDIR
-        self.os_release_file = os_release_file or \
-            os.path.join(self.etc_dir, _OS_RELEASE_BASENAME)
+        self.usr_lib_dir = os.path.join(root_dir, 'usr/lib') \
+            if root_dir else _UNIXUSRLIBDIR
+
+        if os_release_file:
+            self.os_release_file = os_release_file
+        else:
+            etc_dir_os_release_file = os.path.join(
+                self.etc_dir, _OS_RELEASE_BASENAME
+            )
+            usr_lib_os_release_file = os.path.join(
+                self.usr_lib_dir, _OS_RELEASE_BASENAME
+            )
+
+            # NOTE: The idea is to respect order **and** have it set
+            #       at all times for API backwards compatibility.
+            if os.path.isfile(etc_dir_os_release_file) \
```

Comment:
If there is no `/etc/os-release` and no `/usr/lib/os-release` let's set `self.os_release_file` back to the `/etc/os-release` path for backwards compatibility?

Reply:
I checked the if-guard again and what you're asking for is already happening. Am I missing something?

Reply:
You're right! Misread the conditions.